docs/proposals/L7-traffic-route.md

```diff
@@ -0,0 +1,174 @@
+# L7 Traffic Routing
+
+## Context
+
+Kuma provides traffic management with `TrafficRoute` policy based on Kuma tags.
+We can split the traffic to different destinations, we can redirect the traffic etc.
+
+## Requirements
+
+Provide a way to manage the traffic based on HTTP headers, path and method.
+
+## Configuration
+
+```yaml
+type: TrafficRoute
+name: route-all-default
+mesh: default
+sources:
+- match:
+    kuma.io/service: web
+destinations:
+- match:
+    kuma.io/service: backend
+conf:
+  http: # a new http section which is a list
+  - match: # match is required and need to have at least one sub-element
+      method: GET
+      path: # one of either prefix, exact or regex will be allowed
+        prefix: /users
+        exact: /users/user-1
+        regex: 
+      headers:
+        some-header: # one of either prefix, exact or regex will be allowed
+          exact: some-value
+          prefix: some-
+          regex
```

Comment:
Gateway API just moved away from this pattern, to a style that is more consistent with Kubernetes API guidelines:
```
headers:
- name: some-header
  value: some-value
  matchType: exact
```

Similar reasons as for path matches.

Reply:
Here's the kubernetes rationale for avoiding maps, https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#lists-of-named-subobjects-preferred-over-maps